Feature/1061 clause enhancements

testing-life opened the pull request:
## What was done
![image](https://user-images.githubusercontent.com/85691163/180383146-121ca38d-820c-4174-b0ba-1423b086ffac.png)
![image](https://user-images.githubusercontent.com/85691163/180383201-16ac69b7-52d4-4be1-a5c2-f31ba1f223f6.png)

## Testing

#### Before

#### After

hiaux0 commented:
## Fnality

Test text:

**bold**, *italic*, underline

- unoreder list
    - sub
1. orede
    1. sub


### copy paste
- from notion misses underline (note, pasting the notion text into github/gdoc also misses underline, so we might be okay?)

![image](https://user-images.githubusercontent.com/30693990/180719600-e0a750c4-eb66-4045-9a36-36ab9a5fd3d8.png)
![image](https://user-images.githubusercontent.com/30693990/180719509-a95a4ddd-405e-45e5-b870-e9a4ed3f93e9.png)

- from gdoc --> not working

- [ ] Find out if copy paste from other sources should be supported (because "people likely write up the terms somewhere else and just copy into the wizard cc @blink-km)
  - [ ] which ones?

hiaux0 commented:
Could you add into the initial comment 
1. why ckeditor was chosen?
2. what format the formatted text is saved with?

blink-km commented:
copy paste from other sources should be supported

giladt commented on src/resources/elements/primeDesignSystem/pform-input/pform-input.scss:
```diff
@@ -1,4 +1,4 @@
-@import "../variables";
+@import '../variables';
```
Our convention is using double quotes...

testing-life replied:
sorry, automatic formatting

giladt commented on src/styles/styles.scss:
```diff
@@ -1,12 +1,12 @@
-@import "./variables";
-@import "tippy.js/dist/tippy.css"; // TODO: use CDN?
-@import "the-new-css-reset/css/reset.css";
+@import './animations';
```
Same, please keep the double qoutes.

giladt commented on src/resources/elements/primeDesignSystem/pform-input/pform-input.html:
```diff
@@ -2,14 +2,10 @@
   </span>
   <div class="formInputLabelExtra" if.bind="showCounter">
-    ${inputReference.value.length || 0}${maxLength ? '/' + maxLength : ''}
+    ${charValue || inputReference.value.length || 0}${maxLength ? '/' + maxLength : ''}
```
Please note that you're changing a design-system element here. Though integrated with this project, it might be used as a reference in other projects. Is this change applied to the DS or is it specific to the clause text area? If the latter- it may be best to implement it outside of the DS and make sure it's not implicating any of the currently used elements...

@hiaux0 - what's your opinion here?

testing-life replied:
my idea was to extend, which means other instances shouldn't be affected. if you don't pass `charValue` or  `isVertical`, then nothing happens. but if you are unconvinced, by implementing it outside of DS, you mean just making a separate component specifically for the clause fnality ?

giladt commented on src/resources/elements/primeDesignSystem/pform-input/pform-input.html:
```diff
@@ -18,7 +14,7 @@
 
 <!-- Input section -->
-<div class="inputContainer">
+<div class="inputContainer ${isVertical ? 'isVertical':''}">
```
same as above

giladt reviewed (changes requested):
Additionally:
### Styling:
The blue frame for focused inputs does not apply to our design system. It should be  #a258a7 ($Secondary02). Preferably it should be for the whole component (including the menubar):
<img src="https://user-images.githubusercontent.com/2517870/180744851-d6342dd8-337b-4094-a423-4fd5fc5846e5.png" width="300px">

Should be:
<img src="https://user-images.githubusercontent.com/2517870/180748117-e50e1388-21e8-49ee-877d-4c3307be74ff.png" width="300">

The same goes for the `link` box:
<img src="https://user-images.githubusercontent.com/2517870/180748932-0f57bf55-0a75-4bbf-adb2-c3df01134a66.png" width="300">

The border and buttons should match the DS. Currently, we do not have matching icons, but probably colors should do the work. (@v0ort )

testing-life commented:
changes addressed:

- link formatting focus and icons styling
- paste in from gdocs
- initial input height
- input resizing

testing-life commented:
>    why ckeditor was chosen?
documentation, freshness of packages, ease of styling, available plugins. oh and i got it working the easiest :P

>   what format the formatted text is saved with?
it's a string containing the text and formatted words wrapped in HTML elems (i,u,b, ul>li, etc)

giladt commented on src/wizards/tokenSwapDealWizard/stages/termsStage/termClause/custom.css:
```diff
@@ -0,0 +1,118 @@
+  /* -- Overrides generic colors. ------------------------------------------------------------- */
+
+  --ck-color-base-foreground: var(--ck-custom-background);
+  --ck-color-focus-border: #a258a7;
```
please use DS variables ($Secondary02)
You can find it under
/src/resources/elements/primeDesignSystem (/colors.scss)

giladt commented on src/wizards/tokenSwapDealWizard/stages/termsStage/termClause/custom.css:
```diff
@@ -0,0 +1,118 @@
+  --ck-color-button-default-active-shadow: hsl(270, 2%, 23%);
+  --ck-color-button-default-disabled-background: var(--ck-custom-background);
+
+  --ck-color-button-on-background: var(--ck-color-button-active-bg);
```
The same color goes here ($Secondary02)
Text color should be $BG04

giladt commented on src/wizards/tokenSwapDealWizard/stages/termsStage/termClause/custom.css:
```diff
@@ -0,0 +1,118 @@
+  /* -- Overrides the default .ck-toolbar class colors. --------------------------------------- */
+
+  --ck-color-toolbar-background: var(--ck-custom-background);
+  --ck-color-toolbar-border: var(--ck-custom-border);
```
border-color should be $Border01, 
background-color should be $BG04.

giladt reviewed (changes requested):
1. Probably best if you can use the palette colors from the design system for all elements rather than fixed `HSL`, `hex` etc. Please note that there are also variables for sizes, such as border-radius and padding that you can use.

2. I get an error after trying to open a link I've entered. Maybe you can find a fix for that?

3. Please make the hover-border change through the whole element and not only the lower half. If it requires a longer time to refactor, maybe it's better to triage it because it would probably come back to us during testing.

4. Please revert the quotes in all files to double quotes... It appears to apply all `.scss` files.

5. Please don't hate me for being picky! 😂

testing-life commented:
> 1. Probably best if you can use the palette colors from the design system for all elements rather than fixed `HSL`, `hex` etc. Please note that there are also variables for sizes, such as border-radius and padding that you can use.
> 
>     2. I get an error after trying to open a link I've entered. Maybe you can find a fix for that?
>
>     3. Please make the hover-border change through the whole element and not only the lower half. If it requires a longer time to refactor, maybe it's better to triage it because it would probably come back to us during testing.
> 
>     4. Please revert the quotes in all files to double quotes... It appears to apply all `.scss` files.
> 
>     5. Please don't hate me for being picky! joy

ad1. fixed

ad2.
resolved
![image](https://user-images.githubusercontent.com/85691163/180956356-adc8ba34-3a8d-4910-bb5c-58d6ef4a2971.png)

ad3. fixed
![image](https://user-images.githubusercontent.com/85691163/180971356-293c8401-ae15-4240-a4b7-9cf81f9dbfc4.png)

ad4.fixed

ad5. that's ok, i am capable of hating people for several things simultaneously :P

giladt reviewed (changes requested):
Found three more fun tasks for you (While adding a second clause):
1. Styling is gone here:
<img src="https://user-images.githubusercontent.com/2517870/181013956-2ce5086c-1b52-4231-bb15-dee00a5b9788.png" width="300" />


3. IMO we can remove unneeded `/build/translations` files from the repo unless there's a good reason for leaving them there.

4. This:
Please test also deleting and editing saved clauses, to see if it plays right...

<img src="https://user-images.githubusercontent.com/2517870/181009585-530c6f55-9177-4f98-81cf-67bb9e25a29f.png" width="600" />

The rest seems fine with me. Noice work!

testing-life commented:
ad1.  reinstated custom css. overriding some custom props from the component turned out impossible, plus i do think we should look into custom props grolbally, as per the telegram chat
ad2. removed
ad3. `querySelector` now uses index, so displays multiple editors.
